build/templates/ctypes_library.py.mako

```diff
@@ -19,15 +19,17 @@ functions = helper.add_all_metadata(functions)
 %>\
 
 import ctypes
+import threading
 
 from ${module_name}.ctypes_types import *  # noqa: F403,H303
 import ${module_name}.python_types
 
 
 class ${module_name.title()}CtypesLibrary(object):
     def __init__(self, library_name, library_type):
+        self._func_lock = threading.Lock()
         # We cache the cfunc object from the ctypes.CDLL object
-% for f in functions:
+% for f in helper.sorted_functions(functions):
```

Comment:
No, it's not required.

It is because iterating over dictionaries is [non deterministic](https://stackoverflow.com/questions/14956313/why-is-dictionary-ordering-non-deterministic)

By making this change, there won't be arbitrary changes to generated files because Python chose to iterate in a different order.